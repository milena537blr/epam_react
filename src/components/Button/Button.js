import React from "react";
import PropTypes from "prop-types";
import s from "./Button.module.scss";
// import Box from "../Box/Box";
// import classNames from "classnames";

class Button extends React.Component {
  render() {
    return (
     <div>ddd</div>
    );
  }
}

Button.propTypes = {
  text: PropTypes.string
};

export { Button };