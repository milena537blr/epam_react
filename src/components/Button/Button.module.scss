@import "../../styles/core.scss";

.button {
  cursor: pointer;
  padding: rem(10) rem(10);
  border-radius: rem(2);
  font-size: rem(16);
  text-transform: uppercase;
  text-align: center;

  &.white {
    background-color: $color--white;
    color: $color--red;

    &:hover {
      background-color: darken($color--white, 10%);
    }
  }

  &.gray {
    background-color: $color--gray;
    color: $color--white;

    &:hover {
      background-color: lighten($color--gray, 10%);
    }
  }

  &.red {
    background-color: $color--red;
    color: $color--white;

    &:hover {
      background-color: lighten($color--red, 10%);
    }
  }

  &.medium {
    padding: rem(5) rem(20);
    min-width: 100px;
  }

  &.large {
    padding: rem(10) rem(30);
    min-width: 150px;
  }

  &.active {
    font-weight: bold;
    cursor: default;
  }
}
