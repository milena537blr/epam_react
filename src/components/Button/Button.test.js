import React from "react";
import { mount } from "enzyme";
import { Button } from "./Button.js";
import { render } from "@testing-library/react";

const button1 = {
  text: "Medium white button",
  size: "medium",
  color: "white"
};

const button2 = {
  text: "Large gray button",
  size: "large",
  color: "gray"
};

const button3 = {
  text: "Red button",
  color: "red"
};

test("snapshot test", () => {
  const component = mount(<Button />);
  expect(component).toMatchSnapshot();
});

test("generates text, size and color", () => {
  const { getByText, rerender } = render(
    <Button text={button1.text} size={button1.size} color={button1.color} />
  );

  expect(getByText(button1.text)).toBeInTheDocument();
  expect(getByText(button1.text)).toHaveClass(button1.size);
  expect(getByText(button1.text)).toHaveClass(button1.color);

  rerender(
    <Button text={button2.text} size={button2.size} color={button2.color} />
  );

  expect(getByText(button2.text)).toBeInTheDocument();
  expect(getByText(button2.text)).toHaveClass(button2.size);
  expect(getByText(button2.text)).toHaveClass(button2.color);

  rerender(<Button text={button3.text} color={button3.color} />);

  expect(getByText(button3.text)).toBeInTheDocument();
  expect(getByText(button3.text)).toHaveClass(button3.color);
});
