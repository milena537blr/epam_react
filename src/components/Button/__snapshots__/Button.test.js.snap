// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<Button>
  <button
    className="button"
    type="button"
  />
</Button>
`;
